File: arrays/javascript/rotate2dArray.js
var rotate = function(matrix) {
    const dup = JSON.parse(JSON.stringify(matrix))
    for (let row = 0; row < matrix.length; row++) {
        let reverse = matrix.length
        for (let col = 0; col < matrix.length; col++) {
            reverse--
            matrix[row][col] = dup[reverse][row]
        }
    }
    return matrix
};

File: arrays/ruby/rotate2dArray.rb
def rotate(matrix)
#     36%
    duplicate = Marshal.load(Marshal.dump(matrix))
    duplicate.each_with_index do |row, row_index|
        reverse = matrix.length
        row.each_with_index do |column, col_index|
            reverse-=1
            matrix[row_index][col_index] = duplicate[reverse][row_index]
        end
    end
#     17%
    # matrix.replace(matrix.reverse.transpose)
end
